src/Microsoft.ML.Tokenizers/Model/EnglishRoberta.cs

```diff
@@ -233,17 +232,77 @@ public override IReadOnlyList<Token> Tokenize(string sequence)
                 return Bpe.EmptyTokensList;
             }
 
-            IReadOnlyList<Token>? hit = _cache.Get(sequence);
-            if (hit is not null)
+            if (_cache.TryGet(sequence, out IReadOnlyList<Token>? hit))
             {
                 return ModifyTokenListOffsets(hit, indexMapping);
             }
 
-            IReadOnlyList<Token> result = BpeToken(token.Slice(0, newTokenIndex), indexMapping);
+            IReadOnlyList<Token> result = EncodeToTokens(token.Slice(0, newTokenIndex), indexMapping);
             _cache.Set(sequence, result);
             return result;
         }
 
+        /// <summary>
+        /// Tokenize a split sequence string to a list of Ids and add them to the accumulatedIds list.
+        /// </summary>
+        /// <param name="sequence">The sequence to split.</param>
+        /// <param name="isSpecialToken">Indicate if the token is a special token.</param>
+        /// <param name="accumulatedIds">The list of accumulated tokenized Ids.</param>
+        public override void TokenizeToIds(string sequence, bool isSpecialToken, IList<int> accumulatedIds) => TokenizeToIds(sequence, accumulatedIds);
```

Comment:
When do we use the word Tokenize vs the word Encode?

Reply:
We use the `Encode` name at the Tokenizer class level to process the entire input text. At the level of the tokenizer's models, we use the `Tokenize` name, which operates on smaller, pre-tokenized text units.